pkg/controller/deployment/recreate.go

```diff
@@ -31,7 +31,7 @@ func (dc *DeploymentController) rolloutRecreate(d *apps.Deployment, rsList []*ap
 	if err != nil {
 		return err
 	}
-	allRSs := append(oldRSs, newRS)
+	allRSs := append(append([]*apps.ReplicaSet{}, oldRSs...), newRS)
```

Comment:
Would it be better to pre-allocate the array with `make` capacity=len(oldRs), then declaring it inline, and then 2 appends for oldRSs and newRS?

Reply:
done